app/models/wishlist_item.rb

```diff
@@ -27,6 +28,28 @@ def must_fit_in_wishlist_size
     errors.add(:wishlist, I18n.t('wishlist_items.too_many_items'))
   end
 
+  # Validate that domain has supported tld
+  def valid_tld
+    tlds = Setting.find_by(code: 'wishlist_supported_tld')
+    unless tlds.nil?
+      tlds = tlds.value.split('|')
+      current_tld = domain_name.split('.', 2).last
+      return if tlds.include?(current_tld)
+    end
+
+    errors.add(:domain_name, I18n.t('is_invalid')) if errors[:domain_name].blank?
```

Comment:
It may be better to write `errors.add(:domain_name, :key_to_be_translate)`, and add corresponding translation.